src/NServiceBus.Core/InitializableEndpoint.cs

```diff
@@ -141,13 +142,32 @@ void RegisterContainerAdapter(IContainer containerToAdapt)
                 }
 
                 var username = GetInstallationUserName();
+                await CreateQueuesIfNecessary(receiveInfrastructure, username).ConfigureAwait(false);
```

Comment:
While this was run as an installer before, I'd prefer to have this as a separate method called from `Initialize` directly instead. That would eliminate the need to pass `receiveInfrastructure` into `Runstallers`, which otherwise doesn't need it.

Having the call in `Initialize` would also mean the check for send-only could be moved up and be used to skip the call to `CreateQueuesIfNecessary` and `ConfigureReceiveInfrastructure`.